docs/design/per-op-config.md

```diff
@@ -0,0 +1,179 @@
+
+```kotlin
+interface ServiceClient : SdkClient {
+    companion object {
+        fun invoke(block: Config.Builder.() -> Unit): ServiceClient { /* returns instance of implementation class DefaultServiceClient */ }
+    }
+    
+    class Config private constructor(builder: Builder) {
+        val logMode: SdkLogMode = builder.logMode ?: SdkLogMode.Default
+        val httpEngine: HttpClientEngine = builder.httpEngine
+        
+        class Builder {
+            var logMode: SdkLogMode? = null
+            val httpEngine: HttpClientEngine? = null
+            fun build(): Config = Config(this)
+        }
+    }
+    
+    // suspend fun operationA, operationB, etc...
+}
+
+class DefaultServiceClient(config: ServiceClient.Config) : ServiceClient { /* ... */ }
+```
+
+Client interfaces receive a new `withConfig` extension, which takes a config builder and returns a "child" client with
+any fields specified in the builder applied to its config.
+
+```kotlin
+fun ServiceClient.withConfig(block: ServiceClient.Config.Builder.() -> Unit): ServiceClient { /* ... */ }
+```
+
+In practice, to perform one or more operations with a modified config in-place, it is most effective to make your call
+within a `use()` block on the returned client to have the `close()` handled for you. It's still possible to pass the
+subclient to an external background routine, and it may even outlive the parent client (see below).
+
+The `withConfig` method is _more efficient_ than creating multiple clients if you are using runtime-managed `Closeable`
+resources such as the default HTTP engine. Users of these defaults are tracked across clients as they are created, and
+the underlying `Closeable` is only fully closed when the last remaining client using it is closed.
+
+For user-owned `Closeable`s, nothing changes - clients do not handle closing these for you today, and any child clients
+created by `withConfig` won't either.
+
+Sample usage:
+```kotlin
+suspend fun main() {
+    val client = ServiceClient { }
+    
+    client.operationA { /* ... */ }
+    
+    client.withConfig {
+        logMode = SdkLogMode.LogRequestWithBody
+    }.use { // for one-off or explicitly-scoped operation(s), a use {} block is most convenient
+        it.operationA { /* ... */ }
+    }
+    
+    launch {
+        doBackgroundRoutine(client.withConfig {
+            logMode = SdkLogMode.LogResponse
+        })
+        
+        doBackgroundRoutine2(client.withConfig {
+            httpEngine = YourHttpClientEngineImpl()
+        })
+    }
+    
+    // application continues...
+}
+```
+
+Some observations about the above sample:
+* Since no explicit HTTP engine is passed to construction of the root client, the SDK creates its own "managed" default.
+* The first use of `withConfig` and `use` demonstrates how to make basic one-off calls. The defaulted HTTP engine is
+  inherited by the child. When the service client is `closed()` as part of the `use { }` call, the runtime recognizes
```

Comment:
**Nit**: `closed()` isn't a real method name...this should just be "closed" in normal text (without quotes).